doc/source/reference/sparse.migration_to_sparray.rst

```diff
@@ -0,0 +1,268 @@
+.. _migration_to_sparray:
+
+Migration from spmatrix to sparray
+==================================
+
+This document provides guidance for converting code from sparse matrices
+to sparse arrays in SciPy Sparse.
+
+The change from sparse matrices to sparse arrays mirrors conversion from
+``np.matrix`` to ``np.array``. Essentially we must move from an all-2D
+matrix multiplication centric ``matrix`` object to a 1D or 2D “array”
+object that supports operand matrix multiplication and elementwise
+computation.
+
+Notation: For this guide we denote the sparse array classes generally as
+``sparray`` and the sparse matrix classes ``spmatrix``. Dense numpy
+arrays are denoted ``np.array`` and dense matrix classes are
+``np.matrix``. Supported sparse formats are denoted BSR, COO, CSC, CSR,
+DIA, DOK, LIL and all formats are supported by both sparray and
+spmatrix. The term ``sparse`` refers to either ``sparray`` or
+``spmatrix``, while ``dense`` refers to either ``np.array`` or
+``np.matrix``.
+
+Overview and big picture:
+-------------------------
+
+-  the constructor names ``*_matrix``, e.g. ``csr_matrix``, are changed
+   to ``*_array``.
+-  spmatrix ``M`` is always 2D (rows x columns) even
+   e.g. ``M.min(axis=0)``. sparray ``A`` can be 1D or 2D. Numpy scalars
```

Comment:
Perhaps missing something like:

> sparray preserves NumPy semantics, so that ``M.min(axis=0)`` returns a 1D result.

Reply:
Not sure what it means to say "Numpy scalars are used as 0D values where needed."

Reply:
I included an example to make it more clear:
```
Numpy scalars are returned for full reductions, i.e. ``M.min()``.
```